libraries/AP_NavEKF3/AP_NavEKF3_Measurements.cpp

```diff
@@ -521,8 +521,8 @@ void NavEKF3_core::readGpsData()
                 // and set the corresponding variances and covariances
                 alignMagStateDeclination();
 
-                // Set the height of the NED origin to ‘height of baro height datum relative to GPS height datum'
-                EKF_origin.alt = gpsloc.alt - baroDataNew.hgt;
+                // Set the height of the NED origin
+                ekfGpsRefHgt = (double)0.01 * (double)gpsloc.alt + (double)outputDataNew.position.z;
```

Comment:
Is it obligatory to have ekfGpsRefHgt as double ? as it is only use for loc.alt (int32_t) and gpsDataNew.hgt (float), it should prevent unessary casting and double operation, no?